tensorflow/python/ops/embedding_ops.py

```diff
@@ -1007,9 +1009,27 @@ def embedding_lookup_sparse_impl(
     name,
 ):
   """Implementation of sparse embedding aggregation."""
-  if len(params) == 1 and max_norm is None and allow_fast_lookup:
+  need_sparse_segment_gradient = False
+  # Ensure we can query the devices below.
+  segment_ids = ops.convert_to_tensor(segment_ids, name="segment_ids")
+  if (len(params) == 1 and
+      not isinstance(params[0], (resource_variable_ops.BaseResourceVariable,
```

Comment:
Could you use `core.Tensor` and `composite_tensor.CompositeTensor`?  That should cover these two cases, and more.

Or, there is a `tensor_util.is_tf_type(x)` function that covers more types.

Reply:
Thanks, `is_tf_type` didn't work but the first suggestion did. I got the test to pass now. (Also rebased to fix merge conflict).